lib/zhr_devs/identity_management/event_handler.ex

```diff
@@ -0,0 +1,39 @@
+defmodule ZhrDevs.IdentityManagement.EventHandler do
+  @moduledoc """
+  This module is responsible for handling events.
+  Usually used to perform a side effect, such as sending an email or updating a read model.
+  In our case it will be responsible to maintain our in-memory projection of an identity.
+  """
+
+  require Logger
+
+  use Commanded.Event.Handler,
+    application: ZhrDevs.IdentityManagement.App,
+    name: __MODULE__
+
+  alias ZhrDevs.IdentityManagement.Events.LoggedIn
+  alias ZhrDevs.IdentityManagement.Aggregates.Identity
+
+  def handle(%LoggedIn{} = logged_in, meta) do
+    case ZhrDevs.IdentityManagement.spawn_identity(logged_in) do
+      {:ok, _pid} ->
+        Logger.info(
+          "Spawned new identity projection after successful login for #{inspect(logged_in.hashed_identity)}"
+        )
+
+        :ok
+
+      {:error, :already_started} ->
+        # Since we subscribed to the stream of events for this identity we will handle the event there
+
+        :ok
+
+      other_error ->
```

Comment:
What would happen if we crash here?

Reply:
Commanded gives us flexibility here on how we can handle the errors when handling events. We could return `{:error, any()}` here and this will allow us to handle the error in the `error/3` callback, where we have the following options:
```
You can return one of the following responses depending upon the error severity:

{:retry, context} - retry the failed event, provide a context map, or updated [Commanded.Event.FailureContext](https://hexdocs.pm/commanded/Commanded.Event.FailureContext.html) struct, containing any state to be passed to subsequent failures. This could be used to count the number of failures, stopping after too many.

{:retry, delay, context} - retry the failed event, after sleeping for the requested delay (in milliseconds).

:skip - skip the failed event by acknowledging receipt.

{:stop, reason} - stop the event handler with the given reason.
```